compiler/src/dotty/tools/dotc/CompilationUnit.scala

```diff
@@ -105,7 +106,7 @@ class CompilationUnit protected (val source: SourceFile) {
 
 object CompilationUnit {
 
-  class SuspendException extends Exception
+  class SuspendException extends Exception with NoStackTrace
```

Comment:
Adding `NoStackTrace` won't hurt but won't help either for performance. I would `SuspendException` to be really rare.